addons/mail/static/src/core/web/messaging_menu_quick_search.js

```diff
@@ -0,0 +1,15 @@
+import { Component, useState } from "@odoo/owl";
+
+import { useAutofocus, useService } from "@web/core/utils/hooks";
+
+export class MessagingMenuQuickSearch extends Component {
+    static components = {};
+    static props = ["onClose"];
+    static template = "mail.MessagingMenuQuickSearch";
+
+    setup() {
+        super.setup();
+        this.store = useState(useService("mail.store"));
+        useAutofocus();
+    }
+}
```

Comment:
I would also add a `onClose` on click away. This feels cumbersome to use without. Also this would match the behaviour of other toggleable inputs, e. g. `+` buttons in Discuss sidebar category title.